### PR TITLE
Parse grid spans, use `csstree` for placement parsing

### LINKED ISSUE #6628: parse grid span

### PR DESCRIPTION
**Pull request by ruggi**
**Problem:**

1. Parsing an element placing does not use `csstree`
2. Spans are not parsed nor represented

This is a purely-parsing PR, laying the groundwork to have strategies and interactions play nicely with spanning items.

**Fix:**

1. Use `csstree` to parse grid placement
2. Parse `span` coming form parsing props

Fixes #6628

### CODE REVIEW

File: editor/src/components/canvas/dom-walker.ts [hunk 1]
```diff
@@ -635,57 +637,79 @@ function getGridElementProperties(
...
-      ),
-    ) ??
-    null
-  const adjustedColumnEnd =
-    isCSSKeyword(gridColumnEnd) && gridColumn?.end != null ? gridColumn.end : gridColumnEnd
+  const columnSpan: GridSpan | null = isGridSpan(gridColumn?.start)
+    ? gridColumn.start
+    : isGridSpan(gridColumn?.end)
+    ? gridColumn.end
+    : null
```
Comment (bkrmendy) [inline editor/src/components/canvas/dom-walker.ts:644]: this seems to be unused

File: editor/src/components/canvas/dom-walker.ts [hunk 2]
```diff
@@ -635,57 +637,79 @@ function getGridElementProperties(
...
-      parseGridPosition(container, 'row', 'end', gridRow?.end ?? null, elementStyle.gridRowEnd),
-    ) ??
-    null
-  const adjustedRowEnd = isCSSKeyword(gridRowEnd) && gridRow?.end != null ? gridRow.end : gridRowEnd
+  const rowSpan: GridSpan | null = isGridSpan(gridRow?.start)
+    ? gridRow.start
+    : isGridSpan(gridRow?.end)
+    ? gridRow.end
+    : null
+  const gridRowStart = isGridSpan(gridRow?.start)
```
Comment (bkrmendy) [inline editor/src/components/canvas/dom-walker.ts:686]: this seems to be unused too

File: editor/src/components/canvas/dom-walker.ts [hunk 3]
```diff
@@ -635,57 +637,79 @@ function getGridElementProperties(
...
+        parseGridPosition(
+          container,
+          'column',
+          'end',
+          gridColumn?.end ?? null,
+          elementStyle.gridColumnEnd,
+        ),
+      ) ??
+      null
+  const adjustedColumnEnd = isGridSpan(gridColumn?.end)
```
Comment (bkrmendy) [inline editor/src/components/canvas/dom-walker.ts:674]: I'd put these into a helper because these ternary + ?? operator + higher order function combos are a bit hard to parse

Comment (ruggi) [reply editor/src/components/canvas/dom-walker.ts]: [`1d4df1b` (#6629)](https://github.com/concrete-utopia/utopia/pull/6629/commits/1d4df1b4b41077fa1af058452e365acc2d933d32)

**Review by bkrmendy — APPROVED** [review]

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/inspector/common/css-utils.ts [hunk 4]
```diff
@@ -5902,3 +5915,66 @@ function parseRepeatTimes(firstChild: csstree.CssNode) {
...
+      if (isValidGridDimensionKeyword(firstNode.name)) {
+        return cssKeyword(firstNode.name)
+      }
+
+      // …or a line name, in which case look it up in the template and return its index
+      const targetTracks =
+        axis === 'column' ? container.gridTemplateColumns : container.gridTemplateRows
+      const maybeLineFromName =
+        targetTracks?.type === 'DIMENSIONS'
+          ? targetTracks.dimensions.findIndex((dim) => dim.lineName === firstNode.name)
```
Comment (balazsbajorics) [inline editor/src/components/inspector/common/css-utils.ts:5969]: beware that this is not up to spec for repeated line names! `span lineName` seeks for the first index starting from the placement, just `lineName` I think finds the lastIndex. and if the lineName is used for row-start or column-start, the lookup goes backwards!

**Review by balazsbajorics — APPROVED** [review]
